integration-tests/itest/deploy_test.go

```diff
@@ -0,0 +1,306 @@
+		})
+
+		It("should be in the list of applications in the cluster", func() {
+			list, err := cli.ListApps()
+			Expect(err).NotTo(HaveOccurred())
+			appNames := listAppNames(list)
+			Expect(appNames).To(ContainElement(swissKnifeApp.Name))
+		})
+
+		It("should get to a 'running' status, eventually", func(done Done) {
+			// TODO: check status flag once it's fixed
+			status, err := checkStatusIs(swissKnifeApp, "Running")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(status).To(Equal("Running"))
+			close(done)
+		}, LongTimeout)
+	})
+
+	Context("A deployed test application that uses akka, spark, and flink", func() {
+		It("should contain a spark process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("spark-process"))
+		})
+
+		It("should contain a flink process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("flink-process"))
+		})
+
+		It("should contain an akka process", func() {
+			status, err := cli.Status(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			streamlets := cli.GetStreamlets(&status)
+			Expect(streamlets).To(ContainElement("akka-process"))
+		})
+
+		It("should produce a counter in the raw output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "raw-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the akka output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "akka-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the spark output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "spark-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+		It("should produce a counter in the flink output log", func(done Done) {
+			pod, err := cli.GetSinglePodForStreamlet(swissKnifeApp, "flink-egress")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = checkLastLogsContains(pod, swissKnifeApp.Name, "count:")
+			Expect(err).NotTo(HaveOccurred())
+			close(done)
+		}, LongTimeout)
+
+	})
+
+	Context("A deployed application can be undeployed", func() {
+		It("should undeploy the test app", func(done Done) {
+			err := cli.Undeploy(swissKnifeApp)
+			Expect(err).NotTo(HaveOccurred())
+			err = expectFindApp(swissKnifeApp, false)
```

Comment:
Is there no explicit wait time involved here? Or is it hidden somewhere?

Reply:
Timeout management is provided by the framework. Notice the `func(done Done)` which provides a `channel` to communicate that the test is done. If not, the test will interrupt when the time is over and fail with a timeout. 
The timeout period is specified as a parameter to the call, which you find below :arrow_down: 
`, ShortTimeout)`

Reply:
It is fairly unintuitive to me to know that behind the scenes this is actively a repeating poll that only gets short-circuited by either the expected answer being found or the timeout firing.

It's a minor point but IMO for the reader to understand what is going on here it would be useful to get a hint (from the name of the function) that "polling until" is involved.

Reply:
I Agree. I'll change to something like "PollingFor..."